nixos/modules/programs/dwm-status.nix

```diff
@@ -0,0 +1,76 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.dwm-status;
+
+  order = concatMapStringsSep "," (feature: ''"${feature}"'') cfg.order;
+
+  configFile = pkgs.writeText "dwm-status.toml" ''
+    order = [${order}]
+
+    ${cfg.extraConfig}
+  '';
+in
+
+{
+
+  ###### interface
+
+  options = {
+
+    programs.dwm-status = {
+
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to enable dwm-status user service.
+        '';
+      };
+
+      order = mkOption {
+        type = types.listOf types.str;
```

Comment:
The possible values for the features could be defined as an enum and the type of the `order` could become a list of this enum. This would remove the *magic strings* in this module.

Reply:
Great idea, didn't come up with the idea of an enum!